app/models/megaphone/episode.rb

```diff
@@ -125,6 +125,22 @@ def update!(episode = nil)
       self
     end
 
+    def delete!
+      self.api_response = api.delete("podcasts/#{podcast.id}/episodes/#{id}")
+      delete_sync_log
+      delete_delivery_status
+      self
+    end
```

Comment:
when deleting from megaphone, also need to delete the sync log (otherwise it holds on to the guid, and tries to update a deleted episode), and the delivery status (so it will stop trying to sync when there is nothing in megaphone to sync to)